api/src/main/java/org/openmrs/module/locationbasedaccess/aop/interceptor/EncounterServiceInterceptorAdvice.java

```diff
@@ -88,7 +89,13 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
     }
 
     public static Boolean doesEncounterBelongToGivenLocation(Encounter encounter, String sessionLocationUuid) {
-        Location location = encounter.getLocation();
+        Location location;
+        String encounterAccessType = Context.getAdministrationService().getGlobalProperty(LocationBasedAccessConstants.ENCOUNTER_RESTRICTION_TYPE_ENCOUNTER_LOCATION);
+        if(LocationBasedAccessConstants.ENCOUNTER_RESTRICTION_TYPE_PATIENT_LOCATION ==encounterAccessType){
+            location = LocationUtils.getPersonLocation(encounter.getPatient());
+        } else{
```

Comment:
Do we need this `else` part? Can we assign `encounter.getLocation()` as default return to this function?